### PR TITLE
create e2e tests for logo cloud component

### PR DESCRIPTION
**Pull request by marcell-pal-ds**

### CODE REVIEW

File: tests/content/src/main/content/jcr_root/content/howlite-test/pages/Logo-cloud/.content.xml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                            jcr:primaryType="nt:unstructured"
+                            sling:resourceType="howlite/components/image"
+                            lgImageSrc="/content/howlite-test/assets/portrait.jpg"
+                            showLink="false"/>
+                        <image3
+                            jcr:primaryType="nt:unstructured"
+                            sling:resourceType="howlite/components/image"
+                            lgImageSrc="/content/howlite-test/assets/portrait.jpg"
+                            showLink="false"/>
+                    </logoscloud>
```
Comment (ZofiaZ) [inline tests/content/src/main/content/jcr_root/content/howlite-test/pages/Logo-cloud/.content.xml:101]: let's add empty state as well (to check if placeholder is rendered correctly)

File: tests/end-to-end/tests/logo-cloud.cy.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      .should('have.text', 'Logo cloud');
+
+    cy.getByTestId(paths.logoCloudImage)
+      .click()
+      .find(selectors.overlayName)
+      .should('have.text', 'Image');
+
+    cy.getByTestId(testIds.editIcon).click();
+
+    cy.percySnapshotWithAuth('Logo cloud image dialog');
```
Comment (ZofiaZ) [inline tests/end-to-end/tests/logo-cloud.cy.ts:60]: we don't need to test image component (it should be tested in image.cy instead). I think all we could do here is:
- open logo cloud dialog
- take s percysnapshot
- change grid values to for example: 10 cols width, 1 col offset (let's wait for grid PR to be merged)
- save
- check json file

Comment (marcell-pal-ds) [reply tests/end-to-end/tests/logo-cloud.cy.ts:60]: Alright, I'm waiting for this fix until your grid PR will be merged.

Comment (ZofiaZ) [reply tests/end-to-end/tests/logo-cloud.cy.ts:60]: @marcell-pal-ds I've just merged new grid so we can use offset properties now (smOffset, mdOffset and lgOffset)

File: tests/end-to-end/tests/logo-cloud.cy.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    cy.getByTestId(paths.logoCloud)
+      .click({force: true})
+      .find(selectors.overlayName)
+      .should('have.text', 'Logo cloud');
+
+    cy.getByTestId(testIds.editIcon).click();
+
+    cy.percySnapshotWithAuth('Logo cloud dialog');
+
```
Comment (ZofiaZ) [inline tests/end-to-end/tests/logo-cloud.cy.ts:54]: let's add a TODO comment here - for selecting offset and width from dropdown (once WS is ready with test-ids)

File: tests/end-to-end/tests/logo-cloud.cy.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+};
+
+describe('Logo cloud component', function () {
+  beforeEach(() => {
+    cy.login();
+  });
+
+  it('renders correctly in preview mode', function () {
+    cy.visit('/content/howlite-test/pages/Logo-cloud.html');
+    cy.percySnapshotWithAuth('Logo cloud preview');
```
Comment (ZofiaZ) [inline tests/end-to-end/tests/logo-cloud.cy.ts:31]: we could use dedicated percySnaphotPreview method now (that excludes header and footer), same for editor and dialog. Please see my merged PR: https://github.com/websight-io/howlite/pull/54

**Review by ZofiaZ — APPROVED** [review]

**Pull request merged**